docs/source/advanced/finetuning_scheduler.rst

```diff
@@ -0,0 +1,341 @@
+met. A :class:`~pytorch_lightning.callbacks.finetuning_scheduler.fts.FinetuningScheduler` training session completes
+when the final phase of the schedule has its early stopping criteria met. See
+:ref:`Early Stopping<common/early_stopping:Early stopping>` for more details on that callback's configuration.
+
+.. warning:: The :class:`~pytorch_lightning.callbacks.finetuning_scheduler.fts.FinetuningScheduler` callback is in beta
+    and subject to change.
+
+Basic Example
+=============
+If no finetuning schedule is user-provided,
+:class:`~pytorch_lightning.callbacks.finetuning_scheduler.fts.FinetuningScheduler` will generate a
+:ref:`default schedule<advanced/finetuning_scheduler:The Default Finetuning Schedule>` and proceed to finetune
+according to the generated schedule, using default :class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping`
+and :class:`~pytorch_lightning.callbacks.finetuning_scheduler.fts_supporters.FTSCheckpoint` callbacks with
+``monitor=val_loss``.
+
+.. code-block:: python
+
+    from pytorch_lightning import Trainer
+    from pytorch_lightning.callbacks import FinetuningScheduler
+
+    trainer = Trainer(callbacks=[FinetuningScheduler()])
+
+
+.. _default schedule:
+
+The Default Finetuning Schedule
+===============================
+Schedule definition is facilitated via
+:meth:`~pytorch_lightning.callbacks.finetuning_scheduler.fts_supporters.SchedulingMixin.gen_ft_schedule` which dumps
+a default finetuning schedule (by default using a naive, 2-parameters per level heuristic) which can be adjusted as
+desired by the user and/or subsequently passed to the callback. Using the default/implicitly generated schedule will
+often be less computationally efficient than a user-defined finetuning schedule but can often serve as a
+good baseline for subsquent explicit schedule refinement and will marginally outperform many explicit schedules.
+
+
+.. _specifying schedule:
+
+Specifying a Finetuning Schedule
+================================
+
+To specify a finetuning schedule, it's convenient to first generate the default schedule and then alter the
+thawed/unfrozen parameter groups associated with each finetuning phase as desired. Finetuning phases are zero-indexed
+and executed in ascending order.
+
+1. Generate the default schedule to :paramref:`~pytorch_lightning.trainer.trainer.Trainer.log_dir` with the name
+   (:paramref:`~pytorch_lightning.trainer.trainer.lightning_module`.__class__.__name__)_ft_schedule.yaml
+
+.. code-block:: python
+
+    from pytorch_lightning import Trainer
+    from pytorch_lightning.callbacks import FinetuningScheduler
+
+    trainer = Trainer(callbacks=[FinetuningScheduler(gen_ft_sched_only=True)])
+
+
+2. Alter the schedule as desired.
+
+.. container:: sbs-code
+
+    .. rst-class:: sbs-hdr1
+
+        This boring model has four finetuning phases:
+
+    .. rst-class:: sbs-blk1
+
+    .. code-block:: yaml
+      :linenos:
+      :emphasize-lines: 10
+
+            0:
```

Comment:
Agreed! I had planned to add an epoch-driven option in a subsequent version since I found using earlystopping-driven depth transitions as the dominant use case but I can add in this version. Initially thinking I'll add an optional max-epoch constraint within the default earlystopping-driven transitions with a default of -1 (purely early-stopping driven)

Reply:
I didn't think about this. I think it could be possible to abstract the fine-tuning strategy. EpochFinetuning / EarlyStopping one.

Reply:
True, though I wonder if the utility of an abstract EpochFinetuning class might be limited though given that it would essentially serve as a pointer to a set of epochs. I think I have an approach that is working well and would love if you take a look to see what you think. 

I've just pushed a commit to this PR that allows finetuning phase transitions to now be driven by EarlyStopping criteria, exclusively user-specified epoch transitions or a seamless composition of the two (the default mode). 

I've also enabled parameters to be fully specified or selected via regex. Included is a fair amount of validation and testing to ensure EarlyStopping and epoch-driven transition criteria are sensibly composed in a variety of contexts.

![image](https://user-images.githubusercontent.com/7462936/140994386-c5175453-f0ad-4de7-bee9-2c586628f4a5.png)

Lmk what you think sir! Thanks!